### PR TITLE
VPN-3321 - Update to Qt 6.4.1

### PR DESCRIPTION
**Pull request by brizental**
Opening as a draft to see what explodes.

### CODE REVIEW

File: taskcluster/ci/docker-image/kind.yml [hunk 1]
```diff
@@ -33,35 +33,35 @@ tasks:
...
+            QT_VERSION: 6.4.2
     android-build-x64:
         parent: base
         symbol: I(android_x86_64)
         definition: android-qt6-build
         args:
             ANDROID_ARCH: android_x86_64
-            QT_VERSION: 6.2.4
+            QT_VERSION: 6.4.2
     android-qt-next:
```
Comment (strseb) [inline taskcluster/ci/docker-image/kind.yml:65]: We should move this one from 6.2.4 to something newer then :)

Comment (brizental) [reply taskcluster/ci/docker-image/kind.yml:65]: The next version would be 6.5.0-beta2. Do we want to test beta?

**Comment by oskirby** [comment]
I played around a bit with rebuilding Qt for MacOS last week to enable universal builds for MacOS in PR #5797 that kind of ended in failure. If we do make progress in this PR, please make sure to include my changes to [scripts/utils/qt6_compile.sh](https://github.com/mozilla-mobile/mozilla-vpn-client/pull/5828/files#diff-cda671813d90ad9ebfe0ae9e8ee22c379356454d19f22115fa3eecd1920edb4f) to make sure that we properly enable a multi-architecture build of Qt.

**Comment by brizental** [comment]
> I played around a bit with rebuilding Qt for MacOS last week to enable universal builds for MacOS in PR https://github.com/mozilla-mobile/mozilla-vpn-client/pull/5797 that kind of ended in failure. If we do make progress in this PR, please make sure to include my changes to [scripts/utils/qt6_compile.sh](https://github.com/mozilla-mobile/mozilla-vpn-client/pull/5828/files#diff-cda671813d90ad9ebfe0ae9e8ee22c379356454d19f22115fa3eecd1920edb4f) to make sure that we properly enable a multi-architecture build of Qt.

Thanks for the heads up Owen, I have cherry-picked your commit :)

**Comment by brizental** [comment]
Okay, so QSettings in WASM is [interesting](https://github.com/msorvig/qt-webassembly-examples/blob/master/gui_settings/main.cpp#L8-L11). I had to apply multiple workarounds for the tests to work. I was not very previous about these workarounds. I don't see a point in spending too much time fixing these issues on WASM, which is not a user facing platform. I am happy to file follow up issues to clean these up. Let me know.

**Comment by brizental** [comment]
Note: The taskcluster failures are unrelated.

**Comment by brizental** [comment]
> I see the value of WASM for testing, but we could also attempt something similar but not in WASM e.g. by investing some resources on the developer menu of the app.

A bit of context after speaking to @strseb . WASM is useful for translators, because most don't have the application installed. They also require the latest version possible of the application i.e. main or latest release branch. With WASM it's much easier for them to just get that without the hurdle of reinstalling different versions of the application.

That raises the question, why do translators needs the whole app? Wouldn't strings be enough? Basti pointed out we (VPN engineers) are not the best at providing context for strings, so it is easier for translators to just have the app to get that context. Another option would be to have screenshots, but that could easily get out of hand because it would require screenshots for all screens of the app in all states possible.

**Comment by brizental** [comment]
Going back to draft mode... 

Sorry for the back and forth here I just realized layout is all broken with this update e.g. 

Screenshot 2023-02-07 at 13-14-38 Mozilla VPN - WebAssembly Viewer

**Comment by brizental** [comment]
Note: This is now blocked. We will try and see if the update to 6.3 is less painful and start there.

**Comment by brizental** [comment]
Closing this is favour of https://github.com/mozilla-mobile/mozilla-vpn-client/pull/5892

**Pull request closed without merging**